### PR TITLE
Bundle FFI bridges to expose as packages for other platforms

### PR DESCRIPTION
**Pull request by jamilbk**
This PR refactors the native code from `firezone-apple` and `firezone-android` into this repository and then exposes `connlib` as a platform-native package for those platforms.

This has a few benefits:
- All native code is kept out of the client app repositories
- Client apps can include the binary library as a dependency using the idiomatic dependency management tools for each platform
- Keeping all platforms bridges in a single repository will (hopefully) encourage code reuse and help keep the FFIs consistent across bug fixes and new feature development.

### CODE REVIEW

**Comment by jamilbk** [comment]
@wiresock @conectado I'm still wrapping this up, but I thought this would at least be a start for how we can consolidate and collaborate on cross-platform work. I suspect this will be a very active repo.

I apologize in advance for opening a PR that already needs to be refactored (sorry @conectado!) but this was done quickly in order to unblock the Apple and Android developers.

The big piece I left out of course is the Windows piece. This will require more collaboration with @wiresock on how he would like to organize and split things up. Perhaps I can quickly put together a rough UI and/or WebView to mock the user flow using the `windows` crate if that would be helpful.

File: .github/workflows/build.yml
Comment (jamilbk) [inline .github/workflows/build.yml]:
```suggestion
          cargo clippy -- -D clippy::all
```

File: Cargo.toml
Comment (conectado) [inline Cargo.toml]: update boringtun to latest master, they fixed this 2 days ago.

File: Cargo.toml
Comment (conectado) [inline Cargo.toml]: empty features is redundant.

File: android/src/lib.rs
Comment (conectado) [inline android/src/lib.rs]: Just to not forget to do something similar later:
```suggestion
pub extern "system" fn Java_dev_firezone_connlib_Logger_init(_: JNIEnv, _: JClass) {
    #[cfg(debug_assertions)]
    let level = LevelFilter::Trace;
    #[cfg(not(debug_assertions))]
    let level = LevelFilter::Warn;
    
    android_logger::init_once(
        Config::default()
            // Allow all log levels
            .with_max_level(level)
            .with_tag("connlib"),
    )
}
```

File: android/src/lib.rs
Comment (conectado) [inline android/src/lib.rs:61]: nit
```suggestion
    Box::into_raw(Box::new(session))
```

File: android/src/lib.rs
Comment (conectado) [inline android/src/lib.rs]:
```suggestion
pub unsafe extern "system" fn Java_dev_firezone_connlib_Session_disconnect(
```

I know this will only be called externally but I still think it's a good idea to correctly mark unsafe function (if *mut Session is not actually a Session pointer this is UB [see this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=88e826d9e0bc498526ff38a51b5dcd39)

File: android/src/lib.rs
Comment (conectado) [inline android/src/lib.rs]: same as above:

```suggestion
pub unsafe extern "system" fn Java_dev_firezone_connlib_Session_bump_sockets(
```

File: android/src/lib.rs
Comment (conectado) [inline android/src/lib.rs]: same as above
```suggestion
pub unsafe extern "system" fn Java_dev_firezone_connlib_disable_some_roaming_for_broken_mobile_semantics(
```

File: src/lib.rs
Comment (conectado) [inline src/lib.rs:11]: just a a side note we will want to write our own error types and use that instead of `std::io::Error` here but we can leave that for later.

File: src/platform/mod.rs
Comment (conectado) [inline src/platform/mod.rs:1]: We are using the other module file name convention:

`src/platform.rs`

**Review by conectado — CHANGES_REQUESTED** [review]
Look good just a few minor comments.

File: src/lib.rs
Comment (jamilbk) [reply src/lib.rs:11]: Noted, yeah the types will need a lot of work I'm sure

**Comment by jamilbk** [comment]
@conectado Feedback addressed

**Review by conectado — DISMISSED** [review]
great :smiley:

**Comment by jamilbk** [comment]
Can I get a final approval; had to fix a cache issue

**Review by conectado — APPROVED** [review]
can have final approval as a treat

**Pull request merged**
